### ISSUE TITLE
Design System: color fixes for dark mode for red, blue, green, yellow

### ISSUE THREAD

**Issue by mica000** [blockstack/blockstack-app#4831]
### Added fixes to the mathematical scale and to the use case (semantic) for red, yellow, green, blue. 

**ps: Please ignore the INK scale in this image as Fab is working on this in parallel.**

https://www.figma.com/file/2MLHeIeL6XPVi3Tc2DfFCr/%E2%9D%96-Leather-%E2%80%93-Design-System?type=design&node-id=13228-190471&mode=design&t=4atOHlKRSOr96Vb7-4

![Image](https://github.com/leather-wallet/extension/assets/119523541/21bfb52b-79a2-4992-a575-d5c304a0e618)

**Comment by mica000** [comment]
@fabric-8 this seems to not be relevant anymore since we are adding the transparent pallete?

**Comment by mica000** [comment]
@fbwoolf obs: this task is needed to be completed before implementing tags and call out. @kyranjamie ok to move to dev backlog?

**Comment by fbwoolf** [comment]
I can be point person on the UI project now so you can @mention me instead of Kyran to save him some time. I think this looks good to move fwd. 👍

**Comment by fbwoolf** [comment]
PR is up in mono repo to add these tokens: https://github.com/leather-wallet/mono/pull/51

**Comment by fbwoolf** [comment]
@fabric-8 are you going to update info here once the colors are ready? This is blocking me for moving fwd with some other UI issues assigned to me (ex. Callouts and Tags).

**Comment by fabric-8** [comment]
Update regarding the colors: We've tidied up the colors which you can [view and inspect here](https://www.figma.com/file/2MLHeIeL6XPVi3Tc2DfFCr/%E2%9D%96-Leather-%E2%80%93-Design-System?type=design&node-id=13680%3A15870&mode=dev)

![image](https://github.com/leather-wallet/extension/assets/4112698/bd45b5e1-a985-4242-b6ef-687fa533b415)

There might be some more tweaks/adjustments moving forward (ie we're not entirely sure about the exact overlay values yet), but would love that to be something designers could own and simply adjust/pr right in code. I think that's fairly simple to do and would reduce overhead.

Furthermore, let's use straight values right within the semantic tokens vs. having both the full scale tokens and semantic tokens. We'll probably keep on using the scale tokens in figma, but that's more just for us designers to better understand the full context of any given color, and not necessary for code.

**Comment by fbwoolf** [comment]
@fabric-8 any interest in coordinating our border tokens here too? I think we just made them on-the-fly to make dev easier. Rn, we have these...

```js
borders: {
    action: { value: '1px solid {colors.accent.action-primary-default}' },
    active: { value: '2px solid {colors.accent.border-default}' },
    background: { value: '2px solid {colors.accent.background-primary}' },
    dashed: { value: '2px dashed {colors.accent.component-background-default}' },
    default: { value: '1px solid {colors.accent.border-default}' },
    error: { value: '1px solid {colors.error.label}' },
    focus: { value: '2px solid {colors.accent.action-primary-default}' },
    invert: { value: '1px solid {colors.invert}' },
    subdued: { value: '1px solid {colors.accent.text-subdued}' },
    warning: { value: '1px solid {colors.warning.label}' },
  },
```

**Comment by pete-watters** [comment]
@fabric-8 these are places in the extension where we use `ink` now too. We could do with some help semantically naming these: 

**dev branch**
![Screenshot 2024-02-20 at 05 14 51](https://github.com/leather-wallet/extension/assets/2938440/a4cb55ec-6429-48d3-b3bd-c4e3db297e8c)

I am happy to co-ordinate with you on that. The above search is on the `dev` branch. 

You can see below my `containers` branch where we have more:
![Screenshot 2024-02-20 at 05 18 32](https://github.com/leather-wallet/extension/assets/2938440/a46da1d5-e890-45a7-aab3-34dbd3803291)


For some screens like `onboarding` we need to ignore the theme so we could rename some variants of `ink.X` as 'black` or whatever

**Comment by fbwoolf** [comment]
@pete-watters I'm a bit unclear why you jumped in on my issue with your comment? I am in the process of working on this issue which involves coordinating these changes with the design team.

**Comment by pete-watters** [comment]
> ~@pete-watters I'm a bit unclear why you jumped in on my issue with your comment? I am in the process of working on this issue which involves coordinating these changes with the design team.~
> 
> EDIT: Nvm, I see your comment on the mono repo PR making the color changes. Yep, I will coordinate the changes in the extension using this issue. The semantic tokens changes were just the first step. I'll install the new pkg on a branch and update usage of the tokens in the code.

Yeah I was just saying if we ditch `ink` we need to replace it in a few places where we still need it.

**Comment by fbwoolf** [comment]
@fabric-8 how do you want to handle what @pete-watters is pointing out here?

**Comment by fbwoolf** [comment]
Most of these are likely going to impact his work with onboarding.

**Comment by fabric-8** [comment]
@fbwoolf Was planning to do a session with @pete-watters tomorrow where we'll try to resolve those

Regarding borders: Yes, happy to tokenize those as well! But I'll prob. only be able to allocate some time to look into those  thu or fri, so depends a bit on the timeline 😅

**Comment by fbwoolf** [comment]
> @fbwoolf Was planning to do a session with @pete-watters tomorrow where we'll try to resolve those
> 
Ok, let me know here, or we can all meet together, bc I actually need to make the updates to the branch where I'm swapping out all the old color tokens.

> Regarding borders: Yes, happy to tokenize those as well! But I'll prob. only be able to allocate some time to look into those thu or fri, so depends a bit on the timeline 😅
>
No worries, will come back to it later.

**Comment by fbwoolf** [comment]
@fabric-8 is it correct that now `background-primary` and `background-secondary` flipped so that primary is lighter? I think it used to be the opposite so just confirming...

<img width="501" alt="Screenshot 2024-02-20 at 3 59 45 PM" src="https://github.com/leather-wallet/extension/assets/6493321/265352d4-c78f-4f29-860a-a156acbd2ff7">

**Comment by fabric-8** [comment]
Yes correct — This is something we changed during the approval UX work

**Comment by fbwoolf** [comment]
> Yes correct — This is something we changed during the approval UX work

Where can I look at designs to update what is broken here now?
<img width="1511" alt="Screenshot 2024-02-20 at 7 48 07 PM" src="https://github.com/leather-wallet/extension/assets/6493321/3de44379-6819-43de-b7b2-bd4c0a6ec912">

**Comment by pete-watters** [comment]
@fabric-8 @fbwoolf : I added a comment to the PR [here](https://github.com/leather-wallet/extension/pull/4964#pullrequestreview-1892157203) where I went through all the colours we need to rename. 

There is actually only 4 I can see so I'm not sure if we need a call but I'm happy to do one if needed. Even then two of those are the same hex - `lightModeInk.12` + `darkModeInk.1`
We need to semantically name:
```
lightModeInk: {
    1: { value: '#FFFFFF' },
    12: { value: '#12100F' }, 
}

darkModeInk: {
 1: { value: '#12100F' },
12: { value: '#F5F1ED' },
}
```

There are some other rogue colours like `ink.400` but I don't think they match up to anything and are only used in `highlighter.tsx` / `test-app`. They are already broken now.

`Highlighter` gets used in the `CodeBlock` on the error page  + the stacks contract deploy details which I believe will be re-done as part of approvals UX
